### PR TITLE
[MM-26576] fix flaky TestCreateUserInputFilter test

### PR DESCRIPTION
**Pull request by ashishbhate**
#### Summary

-  fix flaky TestCreateUserInputFilter test by deleting created user between interations

#### Ticket Link

- https://mattermost.atlassian.net/browse/MM-26576

### CODE REVIEW

File: api4/user_test.go [hunk 1]
```diff
@@ -85,20 +85,24 @@ func TestCreateUserInputFilter(t *testing.T) {
...
 		})
 
 		th.TestForSystemAdminAndLocal(t, func(t *testing.T, client *model.Client4) {
-			emailAddr := strconv.Itoa(rand.Intn(1000)) + "+testdomainrestriction@mattermost.com"
-			user := &model.User{Email: emailAddr, Password: "Password1", Username: GenerateTestUsername()}
-			_, resp := client.CreateUser(user)
+			user := &model.User{Email: "foobar+testdomainrestriction@mattermost.com", Password: "Password1", Username: GenerateTestUsername()}
+			u, resp := client.CreateUser(user) // we need the returned created user to use its Id for deletion.
+			CheckNoError(t, resp)
+			_, resp = th.SystemAdminClient.PermanentDeleteUser(u.Id)
```
Comment (isacikgoz) [inline api4/user_test.go:105]:
```suggestion
			_, resp = client.PermanentDeleteUser(u.Id)
```

**Comment by agnivade** [comment]
So if I am understanding this correctly, the issue was that 2 consecutive calls to rand.Intn generated the same number which led to a clash?

We also do the same in L153. Perhaps we should change that too?

File: api4/user_test.go (hunk 1 above)
Comment (ashishbhate) [reply api4/user_test.go]: Done!

**Comment by ashishbhate** [comment]
>So if I am understanding this correctly, the issue was that 2 consecutive calls to rand.Intn generated the same number which led to a clash?

Yep, as far as I can make out. Unlikely but not impossible. I still can't reproduce it locally. 

>We also do the same in L153. Perhaps we should change that too?

Good catch! I can't believe I missed it! :(

**Comment by agnivade** [comment]
> Yep, as far as I can make out. Unlikely but not impossible. 

Wow, okay. I thought PRNGs were supposed to _atleast_ guarantee different results in consecutive calls.

**Comment by isacikgoz** [comment]
@lindy65 This is another flaky test fix, nothing significant to test with QA perspective.

**Comment by ashishbhate** [comment]
@agnivade @isacikgoz apologies for the multiple review pings. Since we're reusing the same email in other subtests (order of running subtests is not guaranteed), I've added more clean up deletes.

**Comment by ashishbhate** [comment]
>Wow, okay. I thought PRNGs were supposed to atleast guarantee different results in consecutive calls.

The email generation pattern `strconv.Itoa(rand.Intn(1000)) + "+testdomainrestriction@mattermost.com"` wasn't used anywhere else in our code. So only this test was generating users with that email id. 

All test runs create a new test DB, so its unlikely that a previous test was polluting the test database. 

So I'm not really sure how there could be rand collision, unless the rand seed was reset to the same value

**Pull request merged**